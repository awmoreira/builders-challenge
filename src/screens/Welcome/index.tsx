import React from 'react';
import { useNavigation } from '@react-navigation/native';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { faArrowRight } from '@fortawesome/free-solid-svg-icons';
import {
  Container,
  WrapperTop,
  WrapperBottom,
  AnyText,
  ButtonWeather,
  AvatarImage,
  FooterImage,
} from './styles';
import useLocation from '~/hooks/useLocation';

import { Avatar, Footer } from '~/assets/images';

const Welcome: React.FC = () => {
  const navigation = useNavigation();
  useLocation();

  function gotToWeather() {
    navigation.navigate('Home');
  }

  return (
    <Container>
      <WrapperTop>
        <AvatarImage source={Avatar} />
        <AnyText size={30} weight="bold" margin="32px 0 23px" align="center">
          App Allan do Tempo
        </AnyText>
        <AnyText size={17} weight="300" margin="0 0 30px" align="center">
          Os dados climáticos da sua localização ao apertar de um botão.
        </AnyText>
        <ButtonWeather
          testID="button-see-weather"
          type="primary"
          label="ver previsão do tempo"
          onPress={gotToWeather}
          radius="32px"
          icon={<FontAwesomeIcon icon={faArrowRight} size={14} />}
        />
      </WrapperTop>

      <WrapperBottom>
        <FooterImage source={Footer} />
      </WrapperBottom>
    </Container>
  );
};

export default Welcome;
