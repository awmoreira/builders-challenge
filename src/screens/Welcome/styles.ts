import styled from 'styled-components/native';
import Button from '~/components/Button';

export const Container = styled.View`
  flex: 1;
  background-color: ${({ theme }) => theme.colors.background};
`;
export const WrapperTop = styled.View`
  flex: 1;
  flex-direction: column;
  align-items: center;
  padding-top: 74px;
  padding-horizontal: 77px;
`;

export const WrapperBottom = styled.View`
  width: 375px;
  height: 271px;
  position: absolute;
  bottom: 0;
`;

interface IAnyText {
  size?: number;
  weight?: string | number;
  margin?: string;
  padding?: string;
  color?: string;
  align?: string;
}

export const AnyText = styled.Text<IAnyText>`
  font-size: ${({ size }) => size ?? 16}px;
  font-weight: ${({ weight }) => weight ?? 500};
  margin: ${({ margin }) => margin ?? 0};
  padding: ${({ padding }) => padding ?? 0};
  font-family: ${({ theme }) => theme.fonts.SourceSansPro[500]};
  color: ${({ color }) => color ?? '#000'};
  text-align: ${({ align }) => align ?? 'left'};
`;

export const AvatarImage = styled.Image`
  height: 104px;
  width: 104px;
  margin-top: 54px;
`;

export const ButtonWeather = styled(Button)``;

export const FooterImage = styled.Image`
  height: 100%;
  width: 100%;
`;
