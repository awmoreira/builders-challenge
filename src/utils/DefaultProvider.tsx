import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import theme from '~/theme';

interface IDefaultProvider {
  children: JSX.Element;
}

const DefaultProvider: React.FC<IDefaultProvider> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export default DefaultProvider;
