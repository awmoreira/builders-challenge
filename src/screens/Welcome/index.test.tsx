import React from 'react';
import renderer from 'react-test-renderer';
import DefaultProvider from '~/utils/DefaultProvider';
import { render, cleanup } from '@testing-library/react-native';
import Welcome from '.';
import 'jest-styled-components';

jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

describe('Loading Indicator Component', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render component', () => {
    const tree = renderer
      .create(
        <DefaultProvider>
          <Welcome />
        </DefaultProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render button with background color white for see wheather', () => {
    const { getByTestId } = render(
      <DefaultProvider>
        <Welcome />
      </DefaultProvider>,
    );
    expect(getByTestId(/button-see-weather/)).toBeDefined();
    expect(getByTestId(/button-see-weather/).props.style.backgroundColor).toBe('#fff');
  });
});
