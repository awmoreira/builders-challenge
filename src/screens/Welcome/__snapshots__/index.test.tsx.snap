// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Indicator Component should render component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#EFB743",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingHorizontal": 77,
          "paddingTop": 74,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/avatar.png",
        }
      }
      style={
        Array [
          Object {
            "height": 104,
            "marginTop": 54,
            "width": 104,
          },
        ]
      }
    />
    <Text
      align="center"
      margin="32px 0 23px"
      size={30}
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 30,
            "fontWeight": "bold",
            "marginBottom": 23,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 32,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "textAlign": "center",
          },
        ]
      }
      weight="bold"
    >
      App Allan do Tempo
    </Text>
    <Text
      align="center"
      margin="0 0 30px"
      size={17}
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SourceSansPro-Regular",
            "fontSize": 17,
            "fontWeight": "300",
            "marginBottom": 30,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "textAlign": "center",
          },
        ]
      }
      weight="300"
    >
      Os dados climáticos da sua localização ao apertar de um botão.
    </Text>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRadius": 32,
          "flexDirection": "row",
          "flexGrow": 1,
          "justifyContent": "center",
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "maxHeight": 60,
          "minHeight": 60,
          "opacity": 1,
          "paddingBottom": 18,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 18,
        }
      }
      testID="button-see-weather"
    >
      <
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [
                8594,
              ],
              "f061",
              "M438.6 278.6l-160 160C272.4 444.9 264.2 448 256 448s-16.38-3.125-22.62-9.375c-12.5-12.5-12.5-32.75 0-45.25L338.8 288H32C14.33 288 .0016 273.7 .0016 256S14.33 224 32 224h306.8l-105.4-105.4c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l160 160C451.1 245.9 451.1 266.1 438.6 278.6z",
            ],
            "iconName": "arrow-right",
            "prefix": "fas",
          }
        }
        size={14}
      />
      <Text
        fontsize={14}
        icon={true}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SourceSansPro-Light",
              "fontSize": 14,
              "paddingLeft": 10,
              "textAlign": "center",
            },
          ]
        }
        type="primary"
      >
        ver previsão do tempo
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "height": 271,
          "position": "absolute",
          "width": 375,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/footer.png",
        }
      }
      style={
        Array [
          Object {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    />
  </View>
</View>
`;
